### PR TITLE
Django 4.0 support

### PR DESCRIPTION
**Pull request by HybridAU**
django.conf.urls.url has been removed in favour of djang.urls.path
https://github.com/django/django/blob/3.2.10/django/conf/urls/__init__.py#L16

django.utils.http.is_safe_url has been removed in favour of django.utils.http.url_has_allowed_host_and_scheme
https://github.com/django/django/blob/3.2.10/django/utils/http.py#L330

### CODE REVIEW

**Comment by frinzekt** [comment]
@HybridAU  thanks for this! I'll take a look into it closely with the Django changelog when i get the time

File: django_saml2_auth_lw/urls.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 app_name = 'django_saml2_auth'
 
 urlpatterns = [
-    url(r'^acs/$', views.acs, name="acs"),
-    url(r'^welcome/$', views.welcome, name="welcome"),
-    url(r'^denied/$', views.denied, name="denied"),
+    path('acs', views.acs, name="acs"),
+    path('welcome', views.welcome, name="welcome"),
+    path('denied', views.denied, name="denied"),
 ]
```
Comment (frinzekt) [inline django_saml2_auth_lw/urls.py:10]: okay, yep this looks correct
https://stackoverflow.com/questions/47947673/is-it-better-to-use-path-or-url-in-urls-py-for-django-2-0

Comment (frinzekt) [reply django_saml2_auth_lw/urls.py:10]: it looks like url is just an alias of `path` anyways https://docs.djangoproject.com/en/3.0/ref/urls/#url as per this

Comment (frinzekt) [reply django_saml2_auth_lw/urls.py:10]: I think you should not remove the regex pattern of this one, so probably use `re_path` as per written in https://docs.djangoproject.com/en/3.0/ref/urls/#url

**Review by frinzekt — COMMENTED** [review]
im still reading it... I can't find the equivalence of `url_has_allowed_host_and_scheme` in changelog references

File: django_saml2_auth_lw/urls.py (hunk 1 above)
Comment (HybridAU) [reply django_saml2_auth_lw/urls.py:10]: all the regex start with `^` (beginning of line) and end with `$` (end of line) and don't really use any fancy regex features, so we can just use `path` without needing to use regex

Comment (frinzekt) [reply django_saml2_auth_lw/urls.py:10]: ahh okay sound good

File: django_saml2_auth_lw/views.py [hunk 2]
```diff
@@ -254,9 +254,9 @@ def signin(r):
 
     # Only permit signin requests where the next_url is a safe URL
     if parse_version(get_version()) >= parse_version('2.0'):
-        url_ok = is_safe_url(next_url, None)
+        url_ok = url_has_allowed_host_and_scheme(next_url, None)
     else:
-        url_ok = is_safe_url(next_url)
+        url_ok = url_has_allowed_host_and_scheme(next_url)
```
Comment (frinzekt) [inline django_saml2_auth_lw/views.py:259]: yeah looks good to me. Referenced with https://github.com/django/django/blob/88ed1c8d08c70fd3e7943fc8383459545f726dcd/django/utils/http.py#L270-L276

**Review by frinzekt — APPROVED** [review]
okay, I've smoke tested this and it worked 
Thanks :)

**Pull request merged**